homeassistant/components/pi_hole/sensor.py

```diff
@@ -48,6 +48,11 @@ def name(self):
         """Return the name of the sensor."""
         return f"{self._name} {self._condition_name}"
 
+    @property
+    def unique_id(self):
+        """Return the unique id of the device."""
+        return f"{self.pi_hole.api.host}-{self.name}"
```

Comment:
Host and name are both not unique properties, as they both can change. https://developers.home-assistant.io/docs/en/entity_registry_index.html#unique-id-requirements

Reply:
Do you have an alternate suggestion on properties to use to generate unique IDs for a component like this? I could not find any other parameters myself that could provide better consistency and uniqueness guarantees, and I noticed that the transmission ‘unique_id’ uses the same format. Is it better to not have these components in the entity registry?

Reply:
It's ok to use `getmac` to get the mac address of the host and combine that with the sensor type ie condition name. If getmac returns `None` we fall back to `None` as unique_id.

Reply:
I actually think in this case, IP address is a better indicator of uniqueness than MAC address is since pihole acts as a DNS server and IP addresses are therefore less likely to change than the underlying hardware, but I take your point. I am going to close this for now as I don't think it makes sense to introduce a dependency to `getmac`just for the purposes of getting a unique ID